src/main/java/org/terasology/scenario/internal/systems/ConvertIntoEntitySystem.java

```diff
@@ -0,0 +1,157 @@
+ * limitations under the License.
+ */
+package org.terasology.scenario.internal.systems;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.registry.In;
+import org.terasology.scenario.components.ScenarioHubToolUpdateComponent;
+import org.terasology.scenario.components.actions.ArgumentContainerComponent;
+import org.terasology.scenario.components.information.ConstBlockComponent;
+import org.terasology.scenario.components.information.ConstComparatorComponent;
+import org.terasology.scenario.components.information.ConstIntegerComponent;
+import org.terasology.scenario.components.information.ConstItemPrefabComponent;
+import org.terasology.scenario.components.information.ConstRegionComponent;
+import org.terasology.scenario.components.information.ConstStringComponent;
+import org.terasology.scenario.components.information.InformationEnums;
+import org.terasology.scenario.components.information.PlayerComponent;
+import org.terasology.scenario.internal.events.ConvertIntoEntityConstantEvent;
+import org.terasology.scenario.internal.events.ConvertIntoEntityEvent;
+import org.terasology.scenario.internal.utilities.ArgumentParser;
+import org.terasology.world.block.BlockManager;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class ConvertIntoEntitySystem extends BaseComponentSystem {
+
+    @In
+    EntityManager entityManager;
+
+    @In
+    AssetManager assetManager;
+
+    @In
+    BlockManager blockManager;
+
+    private Logger logger = LoggerFactory.getLogger(ConvertIntoEntitySystem.class);
+
+    @ReceiveEvent
+    public void onConvertIntoEntityEvent(ConvertIntoEntityEvent event, EntityRef entity, ScenarioHubToolUpdateComponent component) {
+        List<String> constructions = event.getConstructionStrings();
+        Pattern patternMain = Pattern.compile("\\[(.*?)\\]");
+        Matcher matcher = patternMain.matcher(constructions.get(0));
+        matcher.find();
+        String sub = constructions.get(0).substring(matcher.end());
+        EntityRef starterEntity = entityManager.create(assetManager.getAsset(sub, Prefab.class).get());
+        ArgumentParser argumentParser = new ArgumentParser();
+        argumentParser.setEntityManager(entityManager);
+        argumentParser.setAssetManager(assetManager);
+        argumentParser.setBlockManager(blockManager);
+        argumentParser.parseDefaults(starterEntity);

```

Comment:
Possibly we want to have a injectable factory for argument parsers so that the construction is not so much code.